### PR TITLE
DMP-1473 Updated notify template - "Error in Processing Audio" fields added

### PR DESCRIPTION
**Pull request by mmmalik1**
### JIRA link (if applicable) ###

(https://tools.hmcts.net/jira/browse/DMP-1473)

### Change description ###

The notify template "Error in Processing Audio" has been updated to "Error in Processing Audio v2"  and extra data parameters have been added to the  new notify screen.

Following have been updated:

NotificationConstants.java
NotificationApi.java
AudioTransformationServiceImpl.java
Functional test errorProcessingAudioV2() added to GovNotifyServiceTest.java
application.yaml

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by davet1985 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java [hunk 1]
```diff
@@ -315,10 +324,27 @@ private void notifyUser(MediaRequestEntity mediaRequestEntity,
...
+            List<DefendantEntity> defendantList = mediaRequestEntity.getHearing().getCourtCase().getDefendantList();
+            List<String> defendantNames = new ArrayList<>();
+            for (DefendantEntity defendant: defendantList) {
+                defendantNames.add(defendant.getName());
+            }
+
+            String defendants = defendantNames.stream().collect(Collectors.joining(","));
+
+            Map<String, String> templateParams = new HashMap<>();
+            templateParams.put(REQUEST_ID, String.valueOf(mediaRequestEntity.getRequestor().getId()));
```
[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java:336]: Updated

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java [hunk 2]
```diff
@@ -315,10 +324,27 @@ private void notifyUser(MediaRequestEntity mediaRequestEntity,
...
+            templateParams.put(COURTHOUSE, String.valueOf(mediaRequestEntity.getHearing().getCourtCase().getCourthouse()));
+            templateParams.put(DEFENDANTS, defendants);
+            templateParams.put(HEARING_DATE, String.valueOf(mediaRequestEntity.getHearing().getHearingDate()));
+            templateParams.put(AUDIO_START_TIME, String.valueOf(mediaRequestEntity.getStartTime()));
+            templateParams.put(AUDIO_END_TIME, String.valueOf(mediaRequestEntity.getEndTime()));
+
```
[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java:342]: Check on template type added

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -142,6 +142,7 @@ darts:
         audio_request_being_processed: d0890ee6-b3d3-45aa-b53a-fcbd651f8aef
         court_manager_approve_transcript: a8390fa6-3f18-44c0-b224-f59971a5e20a
         error_processing_audio: cb5bc3f6-ae1f-4346-845a-622cf6ad2632
+        error_processing_audio_v2: 707fc9fd-4a64-4503-bf6a-4c6bd7dda1dd
```
[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/resources/application.yaml:147]: Removed

File: src/main/java/uk/gov/hmcts/darts/notification/api/NotificationApi.java [hunk 4]
```diff
@@ -11,6 +11,7 @@ public interface NotificationApi {
 
     enum NotificationTemplate {
 
+        ERROR_PROCESSING_AUDIO_V2("error_processing_audio_v2"),
```
[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/java/uk/gov/hmcts/darts/notification/api/NotificationApi.java:14]: Removed

[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/java/uk/gov/hmcts/darts/notification/api/NotificationApi.java:14]: - The test GovNotifyServiceTest.errorProcessingAudio() will be updated with v2 version.

- The  GOV.UK notify template 'Error in Processing Audio' will also be updated in the Transciber folder and 'Error in Processing Audio v2' will be removed

[the remark this answers is not in the document]
Comment (mmmalik1) [reply src/main/java/uk/gov/hmcts/darts/notification/api/NotificationApi.java]: Updated with the application.yaml file referring the new template id

**Review by davet1985 — APPROVED** [review]

**Pull request merged**